sympy/sets/fancysets.py

```diff
@@ -807,6 +832,48 @@ def __getitem__(self, i):
                 raise IndexError("Range index out of range")
             return rv
 
+    def _eval_imageset(self, f):
+        if not self:
+            return S.EmptySet
+        if not isinstance(f.expr, Expr):
+            return
+        if self.size == 1:
+            return FiniteSet(f(self[0]))
+        if f is S.IdentityFunction:
+            return self
+
+
+        # handle f that is univariate and linear
+        # with Rational constants
+        i = Dummy('i', integer=True)
+        a = Wild('a', exclude=[i])
+        b = Wild('b', exclude=[i])
+        match = f(i).match(a*i + b)
+        if not match:
+            return
+        A = match[a]
+        B = match[b]
+        if abs(A.p) == 1 and abs(B) < 1:
+            return
+
+        # it must always be an integer for each value in self
+        F = f(self.step*i + self.start)  # for i in range(len(self))
+        if F.is_integer:
+            n = self.size
+            step = 1
+            match = F.match(a*i + b)
+            a = A
+            b = B
+            p = abs(a.p)
```

Comment:
if F = a*j/aq + b/bq for j in range(0, n, 1) then whether the F(j) is always an integer or not we can write F' = j/aq + b/bq for j in range(0, a*n, a). And that's good enough. Any range starting at a point other than 0 will be remapped to one starting at 0. 

- [ ] check what happens to a neg step range